pkg/node/connection_pool.go

```diff
@@ -73,14 +75,19 @@ func (pool *ConnectionPool) Disconnect(id storj.NodeID) error {
 }
 
 func (pool *ConnectionPool) disconnect(id storj.NodeID) error {
-	i, ok := pool.items[id]
-	if !ok || i.grpc == nil {
+	conn, ok := pool.items[id]
+	if !ok {
+		return nil
+	}
+
+	ptr := atomic.LoadPointer(&conn.grpc)
```

Comment:
This has a potential double close issue. Imagine multiple goroutines calling this concurrently. When they reach this point, multiple goroutines might get the same connection to disconnect.

Reply:
I think that is what the data race is about right?

Reply:
It fixes the data-race, but it still has a logical race.

Reply:
Actually, there shouldn't be any logical race. The `disconnect()` method is invoked in two places: `Disconnect()` and `DisconnectAll()`. Both methods acquire a lock to the same `pool.mu` mutex before calling `disconnect()`.